### DISCUSSION TITLE
Why does `node` need to install dependencies?

### DISCUSSION THREAD

**Discussion by jylin** [Homebrew/discussions#3607] [Everyday usage] [answered]
The node formula depends on `openssl@1.1`, `libuv` and a bunch of other libraries. Why is it needed to install them when the binaries available from the node website (https://nodejs.org/en/download/current/) seem to work out of the box? You can just download the node-v18.8.0-darwin-x64.tar.gz package of binaries and run them. I am on MacOS.

I'd like to install node without installing a bunch of other formula, which are the dependencies.
Is there a mode for homebrew where I can just install the binaries from the website? I thought that was what bottle are for, but it still installs all the dependencies.

**Comment by Bo98** [comment, marked as answer]
The short answer is the Node _does_ use those libraries but their own binary downloads bundle them (statically). We don't bundle dependencies so that they can be shared across multiple formulae.

> Is there a mode for homebrew where I can just install the binaries from the website?

Homebrew-core (formulae) works on an open-source-only basis so does not ever do this. The bottles are built in a Homebrew environment from the source code only.

Homebrew-cask (casks) does use upstream binaries, though casks work quite differently to formulae in general. Casks offer different guarantees to formulae so are generally aimed for things that aren't suited as a formula, such as GUI apps and closed source software.

You may however find third party taps which offer formulae that use upstream binaries. They are not maintained by Homebrew so there's no global list.
